apps/server/src/modules/authorization/authorization.service.spec.ts

```diff
@@ -1,123 +1,354 @@
+			const role = roleFactory.build({ permissions: ['a'] });
+			const user = userFactory.build({ roles: [role] });
+
+			const permissions = service.resolvePermissions(user);
+
+			expect(permissions).toEqual(['a']);
+		});
+
+		it('should return permissions of a user with many roles', () => {
+			const roleA = roleFactory.build({ permissions: ['a'] });
+			const roleB = roleFactory.build({ permissions: ['b'] });
+			const user = userFactory.build({ roles: [roleA, roleB] });
+
+			const permissions = service.resolvePermissions(user);
+
+			expect(permissions.sort()).toEqual(['a', 'b'].sort());
+		});
+
+		it('should return unique permissions', () => {
+			const roleA = roleFactory.build({ permissions: ['a', 'b'] });
+			const roleB = roleFactory.build({ permissions: ['b', 'c'] });
+			const user = userFactory.build({ roles: [roleA, roleB] });
+
+			const permissions = service.resolvePermissions(user);
+
+			expect(permissions.sort()).toEqual(['a', 'b', 'c'].sort());
+		});
+
+		it('should return the permissions of the 1st level sub role', () => {
+			const roleB = roleFactory.build({ permissions: ['b'] });
+			const roleA = roleFactory.build({ permissions: ['a'], roles: [roleB] });
+			const user = userFactory.build({ roles: [roleA] });
+
+			const permissions = service.resolvePermissions(user);
+
+			expect(permissions.sort()).toEqual(['a', 'b'].sort());
+		});
+
+		it('should return the permissions of nested sub roles', () => {
+			const roleC = roleFactory.build({ permissions: ['c'] });
+			const roleB = roleFactory.build({ permissions: ['b'], roles: [roleC] });
+			const roleA = roleFactory.build({ permissions: ['a'], roles: [roleB] });
+			const user = userFactory.build({ roles: [roleA] });
+
+			const permissions = service.resolvePermissions(user);
+
+			expect(permissions.sort()).toEqual(['a', 'b', 'c'].sort());
+		});
+
+		it('should return unique permissions of nested sub roles', () => {
+			const roleC = roleFactory.build({ permissions: ['c', 'a'] });
+			const roleB = roleFactory.build({ permissions: ['b'], roles: [roleC] });
+			const roleA = roleFactory.build({ permissions: ['a'], roles: [roleB] });
+			const user = userFactory.build({ roles: [roleA] });
+
+			const permissions = service.resolvePermissions(user);
+
+			expect(permissions.sort()).toEqual(['a', 'b', 'c'].sort());
+		});
+
+		it('should throw an error if the roles are not populated', () => {
+			const user = userFactory.build();
+			user.roles.set([orm.em.getReference(Role, new ObjectId().toHexString())]);
+
+			expect(() => service.resolvePermissions(user)).toThrowError();
+		});
+
+		it('should throw an error if the sub-roles are not populated', () => {
+			const role = roleFactory.build();
+			role.roles.set([orm.em.getReference(Role, new ObjectId().toHexString())]);
+			const user = userFactory.build({ roles: [role] });
+
+			expect(() => service.resolvePermissions(user)).toThrowError();
+		});
 	});
+	describe('Authorization', () => {
```

Comment:
The describe('Authorization', can be removed, is already on top and all of the methodes has one describe on layer deeper.